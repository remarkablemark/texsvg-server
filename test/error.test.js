const supertest = require('supertest');
const app = require('../app');
let { contentType } = require('../helpers/constants');
const svgs = require('../helpers/svgs');

jest.mock('texsvg', () => jest.fn().mockRejectedValue('error'));

const textHtml = 'text/html; charset=utf-8';
const status500 = 500;

const agent = supertest.agent(app);

const tex = 'x^2';
describe(`GET /?tex=${tex}`, () => {
  it('catches error', async () => {
    await agent
      .get(`/?tex=${tex}`)
      .expect(contentType, textHtml)
      .expect(status500, status500.toString())
      .expect(res => {
        expect(res.text).toBe('500');
        expect(res.error).toMatchSnapshot();
      });
  });
});
