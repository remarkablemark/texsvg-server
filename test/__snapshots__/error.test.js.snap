// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /?tex=x^2 catches error 1`] = `[Error: cannot GET /?tex=x%5E2 (500)]`;
